geode-assembly/src/upgradeTest/java/org/apache/geode/management/RollingUpgradeWithSslDUnitTest.java

```diff
@@ -0,0 +1,276 @@
+    final List<String> result = VersionManager.getInstance().getVersionsWithoutCurrent();
+    result.removeIf(s -> TestVersion.compare(s, "1.10.0") < 0);
+    return result;
+  }
+
+  @Rule
+  public GfshRule oldGfsh;
+
+  @Rule
+  public GfshRule currentGfsh;
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  public RollingUpgradeWithSslDUnitTest(String version) throws UnknownHostException {
+    oldGfsh = new GfshRule(version);
+    currentGfsh = new GfshRule();
+    hostName = InetAddress.getLocalHost().getCanonicalHostName();
+    keyStoreFileName = hostName + "-keystore.jks";
+    trustStoreFileName = hostName + "-truststore.jks";
+  }
+
+  @Before
+  public void before() throws IOException, GeneralSecurityException {
+    generateStores();
+    /*
+     * We must use absolute paths for truststore and keystore in properties file and
+     * since we don't know those at coding-time, we must generate the file.
+     * Since GfshRule provides no way to pass along Properties object to start server etc,
+     * we must write the properties to an actual file.
+     */
+    final Properties properties = generateSslProperties();
+
+    securityPropertiesFile = tempFolder.newFile("gfsecurity.properties");
+    final FileOutputStream fileOutputStream =
+        new FileOutputStream(securityPropertiesFile.getAbsolutePath());
+    properties.store(fileOutputStream, "");
+  }
+
+  @Test
+  public void testRollingUpgradeWithDeployment() throws Exception {
+    final int locatorPort = portSupplier.getAvailablePort();
+    final int locatorJmxPort = portSupplier.getAvailablePort();
+    final int locator2Port = portSupplier.getAvailablePort();
+    final int locator2JmxPort = portSupplier.getAvailablePort();
+    final int server1Port = portSupplier.getAvailablePort();
+    final int server2Port = portSupplier.getAvailablePort();
+
+    final GfshExecution startupExecution =
+        GfshScript.of(
+            startLocatorCommandWithConfig("locator1", locatorPort, locatorJmxPort, -1))
+            .and(startLocatorCommandWithConfig("locator2", locator2Port, locator2JmxPort,
+                locatorPort))
+            .and(startServerCommandWithConfig("server1", server1Port, locatorPort))
+            .and(startServerCommandWithConfig("server2", server2Port, locatorPort))
+            .execute(oldGfsh, tempFolder.getRoot());
+
+    initializeRegion(locatorPort);
+    causeP2PTraffic(locatorPort);
+
+    // doing rolling upgrades
+    upgradeLocator("locator1", locatorPort, locatorJmxPort, locator2Port, startupExecution);
+    verifyListMembers(locatorPort);
+    causeP2PTraffic(locatorPort);
+
+    upgradeLocator("locator2", locator2Port, locator2JmxPort, locatorPort, startupExecution);
+    verifyListMembers(locatorPort);
+    causeP2PTraffic(locatorPort);
+
+    // make sure servers can do rolling upgrade too
+    upgradeServer("server1", server1Port, locatorPort, startupExecution);
+    causeP2PTraffic(locatorPort);
```

Comment:
This is not checking if the entry was made into server1 and read from server2; i am thinking thats the expectation here...Please correct me if I am missing anything.